File: 49-group-anagrams/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        grouping = defaultdict(list)
        for s in strs:
            k = "".join(sorted(list(s)))
            grouping[k].append(s)
        res = []
        for v in grouping.values():
            res.append(v)
        return res

File: 49-group-anagrams/49-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let grouping = {};
    strs.forEach(str => {
        sorted = str.split("").sort().join("");
        if (grouping[sorted] == undefined){
            grouping[sorted] = [];
        }
        grouping[sorted].push(str);
    })
    return Object.values(grouping);
};